app/models/sponsorship.rb

```diff
@@ -16,6 +16,41 @@ class Sponsorship < ActiveRecord::Base
   belongs_to :bill, required: true
   belongs_to :legislator, required: true
 
-  scope :sponsored,   -> { where.not(introduced_at: nil) }
-  scope :cosponsored, -> { where.not(cosponsored_at: nil) }
+  scope :sponsored,       -> { where.not(introduced_at: nil) }
+  scope :cosponsored,     -> { where.not(cosponsored_at: nil) }
+  scope :pledged_support, -> { where.not(pledged_support_at: nil) }
+  scope :current,         -> { joins(:bill).merge(Bill.current) }
+
+  delegate :name, :congressional_session, to: :bill
+
+  SPONSORSHIP_LEVELS = [
+    [:introduced_at, 'sponsored'],
+    [:cosponsored_at, 'cosponsored'],
+    [:pledged_support_at, 'pledged_support']
+  ]
+
+  def current_sponsorship
+    SPONSORSHIP_LEVELS.each do |field_name, level|
```

Comment:
I feel like this is a little overly complicated and little difficult to immediately grok what is going on.  What about....

```
def current_sponsorship_level
  case
    when introduced_at then :introduced_at
    when cosponsored_at then :cosponsored_at
    when pledged_support then :pledged_support_at
  end
end

def current_sponsorship_at
  introduced_at || cosponsored_at || pledged_support_at
end
```

Both will return `nil` if no value exists and a future developer can see what is going on without having to carefully read the `current_sponsorship` method.